content/blog/2020/07/2020-07-24-custom-distribution-service-phase-2.adoc

```diff
@@ -0,0 +1,189 @@
+the repository or to have them in a completely new repository. 
+Let us talk about both these approaches.
+
+=== Having the configurations in the current repository:
+
+This allows us to have all of the relevant configurations inside the repository itself,
+and so users would not have to go fetch this in different repositories.
+We could have issues with the release cycle and dependencies since,
+it would have to happen along with the custom distribution service project releases.
+
+==== Having the configurations in a different repository:
+
+This allows us to manage all of the configurations and the relevant dependencies separately and easily,
+thus avoiding any release conflict with the current repository.
+However it would be a bit difficult if users were to not find this repository.
+
+**Decision** : We still cannot quite agree on what is the best method so for now,
+I have included the url from which the community configurations are picked up as a
+configuration variable in the **.env** file which can be configured later and
+therefore it can be upto the user to configure. Another advantage of having it configurable,
+is that the user can decide to load configurations which are private to his organization as well. 
+
+**Issue:**   link:https://github.com/jenkinsci/custom-distribution-service/issues/6161]
+
+**Pull Request:**   link:https://github.com/jenkinsci/custom-distribution-service/pull/73[73]
+
+== Back-End
+
+=== War Generation
+
+The ability to generate and download war files has finally been achieved,
+the reason this feature took so long to complete is because we had some difficulty
+in implementing the war generation and its tests. However this has been completed
+and can now be tested successfully.
+
+==== Things to take care while generating war files
+
+In its current state the war generation cannot include **casc.yml** or **groovy** files
+if they are included in the configuration they would have to be added externally.
+The war file generation would yell at you if you tried to build a war file with a jcasc file configuration. 
+
+**Issue:**   link:https://github.com/jenkinsci/custom-distribution-service/issues/60[60]
+
+**Pull Request:**   link:https://github.com/jenkinsci/custom-distribution-service/pull/68[68]
+
+=== Pull Request Creation
+
+This feature was included in the design document that I created after my GSoC selection.
+It involves the ability to create pull requests via the front-end of the service.
+The User Story behind this feature was that **If I want to share a configuration with the community and I do not quite know how to use github or I do not want to do it via the terminal**.
+This feature includes creation of a bot that handles the creation of pull requests in the repository. 
+This bot would have to be installed by the jenkins organization in this repository and the bot would handle the rest. 
+
+==== Disclaimer: 
+
+This feature has however been put on the back-burner for now because
```

Comment:
Note: this might be a good place to say if the reader wants to participate in the discussion that they can look at the PR or join on the gitter channel.